lib/android/app/src/main/java/com/reactnativenavigation/react/ReactView.java

```diff
@@ -12,55 +17,78 @@
-		setEventListener(new ReactRootView.ReactRootViewEventListener() {
+    private ScrollView scrollView;
+    private ViewTreeObserver.OnScrollChangedListener scrollChangedListener;
 
-			@Override
-			public void onAttachedToReactInstance(final ReactRootView reactRootView) {
-				reactRootView.setEventListener(null);
-				isAttachedToReactInstance = true;
-			}
-		});
-		final Bundle opts = new Bundle();
-		opts.putString("containerId", containerId);
-		startReactApplication(reactInstanceManager, containerName, opts);
-	}
+    public ReactView(final Context context, ReactInstanceManager reactInstanceManager, String containerId, String containerName) {
+        super(context);
+        this.reactInstanceManager = reactInstanceManager;
+        this.containerId = containerId;
+        this.containerName = containerName;
+        start();
+    }
 
-	@Override
-	public boolean isReady() {
-		return isAttachedToReactInstance;
-	}
+    private void start() {
+        setEventListener(reactRootView -> {
+            reactRootView.setEventListener(null);
+            isAttachedToReactInstance = true;
+        });
+        final Bundle opts = new Bundle();
+        opts.putString("containerId", containerId);
+        startReactApplication(reactInstanceManager, containerName, opts);
+    }
 
-	@Override
-	public View asView() {
-		return this;
-	}
+    @Override
+    public boolean isReady() {
+        return isAttachedToReactInstance;
+    }
 
-	@Override
-	public void destroy() {
-		unmountReactApplication();
-	}
+    @Override
+    public View asView() {
+        return this;
+    }
 
-	@Override
-	public void sendContainerStart() {
-		new NavigationEvent(reactInstanceManager.getCurrentReactContext()).containerDidAppear(containerId);
-	}
+    @Override
+    public void destroy() {
+        unmountReactApplication();
+    }
 
-	@Override
-	public void sendContainerStop() {
-		new NavigationEvent(reactInstanceManager.getCurrentReactContext()).containerDidDisappear(containerId);
-	}
+    @Override
+    public void sendContainerStart() {
+        new NavigationEvent(reactInstanceManager.getCurrentReactContext()).containerDidAppear(containerId);
+    }
+
+    @Override
+    public void sendContainerStop() {
+        new NavigationEvent(reactInstanceManager.getCurrentReactContext()).containerDidDisappear(containerId);
+    }
+
+    @Override
+    public void addView(View child, int index, ViewGroup.LayoutParams params) {
+        super.addView(child, index, params);
+        if (child instanceof ScrollView) {
+            scrollView = (ScrollView) child;
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    public void setScrollListener(ScrollListener scrollListener) {
+        if (scrollView != null) {
+            if (scrollListener != null) {
+                scrollChangedListener = () -> scrollListener.onScroll(scrollView.getScrollY());
+                scrollView.getViewTreeObserver().addOnScrollChangedListener(scrollChangedListener);
```

Comment:
Last time I checked, this method "skips" scroll events. The events reported are kind of batched or posted, meaning the callback isn't invoked immediately after a scroll. This means that if we need to track finger, this method yields subpar results.

Reply:
From what I've tested, it produces enough events to track finger. I mean we don't need that much precision.